rules.md

```diff
@@ -66,3 +66,221 @@ and fail unexpectedly when supplied extra long strings.
+## B010: Pluralize resource names
+
+Resource names should be pluralised in paths.
+Path components after prefix and version should be plurals, for example, `/client-api/v1/bars` client and v1 are ignored.
+
+## B009: Check prefix for paths should contain version
+
+URL Paths Must contain a version number to maintain a consistent method of versioning though the apis.
+## B008: Check x-icon value in the info block
+
+An x-icon should be provided in the info block of the spec with the assigned value for the API. This should be provided 
+so that the api portal can display the api with the correct icon.
+
+## B007: Check prefix for paths
+
+Path prefixes should be from this set of valid prefixes as it should refer to the type of api.
+
+Valid prefixes:
+ - client-api
+ - service-api
+ - integration-api
+ 
+## B006: Check info block description format
+
+Descriptions are important to include for readability and understanding the purpose of a spec.
+A description must be present in the info block, it cannot be empty and must be no longer then the configured max length 
+which is currently set to 140. 
+
+## B005: Check info block title format
+
+For readability the title must not be empty and should be no longer then the configured max title length, 35.
+
+## B004: Check info block tags allowed
+
+Tags in the info block, are required to group apis together, they must be from the list of valid meaningful tags below 
+to create a consistent, uniform and well understood set of apis.
+The Tags value must not be empty. They must be used exactly as shown below as they are case-sensitive.
+
+Valid info tags:
+
+ - productTags: 
+   - Retail 
+   - Business
+   - Wealth
+   - Identity
+   - Foundation
+   - Basic Support
+   - Flow
+ - informativeTags: 
+   - Mobile
+   - Security
+   - Payments
+   - Authentication
+   - Employee
+   - Cash
+   - Insights
+   
+## B003: No reserved words allowed
+   
+Schema must not contain reserved words as keys, the reserved words from js, spring kotlin and swift, are checked for. 
+This is to avoid confusion, when mapping to objects and to avoid injecting code.
+Some examples are :
+
+- Spring:
+    assert, boolean, class, null, new, public
+- Js:
+    abstract, delete, export, import, volatile, static
+- Kotlin:
+    break, interface, throw, super
+- Swift:
+    \#column, \#line, Character, Data, Int, left
+## B002: All operationIds must be unique
+
+Duplicate operationIds cause confusion especially when generating code from specifications, they must be unique.
+
+## B001: No license information allowed
+
+OpenAPI must not contain a license  because it's covered by the License Agreement we already negotiate with customers.
+
+## M00012: Open API Version must be set to the correct version
+
+OpenAPI specification version must be 3.0.3 or 3.0.4. Any others are not compatible.
+
+
+
+# Enabling linting
+
+To enable linting add the execution goal `lint` to your plugin configuration, this hooks it up to your build process.
```

Comment:
This section might be better structuredas:

To enable linting, add the execution goal `lint` to your plugin configuration. This hooks linting into your build process. For example:
<<add POM example here>>
The following linting parameters are available:
<<add linting parameters here>>